### ISSUE TITLE
Support `CACHEDIR.TAG` for wrap caches

### ISSUE THREAD

**Issue by bgilbert** [mesonbuild/meson#12103]
**Describe the bug**
Similar to #11734, it'd be useful to add `CACHEDIR.TAG` to `subprojects/packagecache` and to subproject directories that have been unpacked from wraps, so backup software will ignore them.

The docs have [language](https://mesonbuild.com/Wrap-dependency-system-manual.html#specific-to-wrapfile) (under "since 0.49.0") suggesting that some projects may manually put files in `packagecache`, which would make it not really a cache in that case.  Should we detect that case (by scanning `source_filename`/`patch_filename` values) and avoid writing `CACHEDIR.TAG` to `packagecache`?

**Comment by eli-schwartz** [comment]
I'm *very* skeptical that a `CACHEDIR.TAG` which is automatically generated in the packagecache directory has advantages over one manually created and committed to the repository.

In contrast:
- the builddir is automatically generated absolutely anywhere, so you cannot commit it to git as that simply doesn't work regardless of whether you want it or not
- the subprojects extraction has a stable name based on the wrap file which is committed to git, but you can't commit just a few of its files and then have meson agree to extract the tarball there too, so it doesn't currently work although in theory meson could add a hardcoded quirk to respect one checked into git.

> The docs have language (under "since 0.49.0") suggesting that some projects may manually put files in `packagecache`, which would make it not really a cache in that case.

The design behind that feature is that you can pre-seed the cache for example if `meson setup` is run without networking. It's also been (ab)used by projects wanting the functionality that we eventually implemented as the `packagefiles` directory, i.e. wraps that have locally maintained meson.build overlays without a remote download url -- so you could put junk data in that field and rely on meson checking for a "cached" copy first. People should just use `packagefiles`, but they may not have updated their minimum meson yet, or may not have refactored since they updated their minimum.

**Comment by tristan957** [comment]
I agree with @bgilbert here. Expecting every project that uses subprojects to include this is not very practical.

**Comment by tristan957** [comment]
I would like to see Meson generate a gitignore for the packagecache directory. I have to add it to every project anyway.
